### DISCUSSION TITLE
Sanity Nextjs Integration

### DISCUSSION THREAD

**Discussion by Collinsmbaka** [vercel/next.js#43235] [Help] [answered]
Hey all,

I’m new to Sanity and trying to deploy a nextjs app with vercel using Sanity

this is the repo I’m trying to deploy
https://vercel.com/templates/next.js/nextjs-portfolio

when I click on deploy and follow the prompt, it asks me to fill up some secret environment variables.
I have created a project on Sanity and was able to get the project ID, API token, and dataset values

what I don’t know how to get is the
“SANITY_PREVIEW_SECRET” and “SANITY_STUDIO_REVALIDATE_SECRET”
These values are needed to connect to Sanity and fetch content for blog posts

how do I get these values?

any help will be appreciated!

thank you

**Comment by icyJoseph** [comment, marked as answer]
Follow this guide:

- https://github.com/vercel/next.js/tree/canary/examples/cms-sanity#add-the-preview-secret-environment-variable
- https://github.com/vercel/next.js/tree/canary/examples/cms-sanity#step-7-setup-revalidation-webhook

These are keys you ought to generate yourself. I think the revalidation hook you ought to read carefully, because you have to go some dashboard and add it. The PREVIEW_SECRET variable is just used by the preview route to make sure that unpublished content isn't publicly available.

**Reply by Collinsmbaka** [reply]
Thank you, Joseph!

That was really helpful
